### ISSUE TITLE
Links with descriptions not linkified in org-ql-view

### ISSUE THREAD

**Issue by jakebox** [alphapapa/org-ql#282]
Situation: I have a large org file with a number of headlines tagged `:article:` (things I might want to read). I like to get all of them at once with a simple org-ql query. I was writing a function to export the buffer so I can have a single file of tags, but it turns out that the links do not transfer because I use the orgmode link syntax in the heading (the [[link][description]]).

Is it possible to have the links be transferred to the org-ql result? Or, at least, have them be highlighted with the org-link face (like links without descriptions are)?

Thank you! Hope this explanation is clear.

**Comment by alphapapa** [comment]
Hi Jake,

I'm sorry, but I will need you to be even more specific before I can help you.

You may also consider that this may be a case of the "XY problem," and that it would probably be helpful if you sought help on, e.g. https://old.reddit.com/r/emacs+orgmode, where many other people can offer help, and others can benefit from the answers.  This tracker is best reserved for bug reports and feature requests (and I think there isn't one here, since AFAIK org-ql already does what you ask for).

**Comment by jakebox** [comment]
Sure. Let's see if I can better explain with some screenshots.

This is my org-agenda, built from the following two headlines:

```
*** TODO [[https://github.com/alphapapa/org-ql/issues/282][A headline with a link]]
*** TODO A headline with an exposed link: https://github.com/alphapapa/org-ql/issues/282
```

![Screen Shot 2022-06-14 at 10 15 11 AM](https://user-images.githubusercontent.com/26366643/173613655-d8929cdf-244f-41e8-bd0c-3ef762db967e.png)

Both of the items are highlighted in the `org-link` face.

If I do an org-ql query on the same file to match all TODOs `(todo "TODO")` I get this:

![Screen Shot 2022-06-14 at 10 17 21 AM](https://user-images.githubusercontent.com/26366643/173614106-3964e0cd-5c3d-4930-a96a-fe6bf32974aa.png)

The first item, a link with a description, has face `default` — that is the issue I am attempting to report/feature I am requesting. I think that the headline, when queried with org-ql, should have the `org-link` face like it does when viewed through org-agenda.

Hope that makes sense. Thanks!

**Comment by alphapapa** [comment]
Now I understand.  That will probably be a simple fix.  I'll plan to fix it in the 0.7 release, but that may not be for a while, as my time to work on org-ql lately is limited.  Thanks for reporting it.

**Comment by jakebox** [comment]
Cool. Thanks! I'll leave the issue open for you to close then.

**Comment by alphapapa** [comment]
@jakebox  Thanks for your patience.  Looking at this issue again, I'm not sure what the correct fix is, because I don't fully understand how, where, and when Org is "prettifying" links.

I noticed that when I comment out this call to `org-link-display-format` in `org-ql-view--format-element`: https://github.com/alphapapa/org-ql/blob/7f68db962e16a47f22bc27bcd95b4d46581e3176/org-ql-view.el#L855 ...it seems to produce the intended appearance.  This is on Emacs 28 with Org 9.5.2.

Could you please test the same thing and see if it fixes it for you?  i.e. do:

1. `C-h f org-ql-view--format-element RET` and click the link to go to the source code.
2. Comment out the call to `org-link-display-format` like so:
```el
           (title (--> (org-ql-view--add-faces element)
                       (org-element-property :raw-value it)
                       ;; (org-link-display-format it)
                       ))
```
3. `C-M-x` to evaluate the `defun`.
4. Do an `org-ql-search` containing links that weren't displayed correctly before, and see if they have the correct appearance now.

**Comment by jakebox** [comment]
Yes, that works for me! I'm on `28.2` with Org version `9.5.5`.

Appearance is correct. Interesting how the link still exists (i.e. I can open it regardless of the formatting with `C-c C-o`). 

Thanks for coming back to this.

**Comment by alphapapa** [comment]
@jakebox Thank you for your patience, and for reporting this.  It should finally be fixed.

I still don't fully understand how or why this fixes it, but since it appears to work fine, I'm not going to spend the time to dig into Org's internals to find out why.  If it turns out that this breaks something else, we'll revisit and fix it.

I released this fix in v0.8.1 and on `master`.

**Comment by jakebox** [comment]
Wonderful, thank you for your diligence! Glad to see this fixed.
